### PR TITLE
Display "Collecting type info for N module(s) ..." message for GHC 8

### PR DESCRIPTION
**Pull request by vshabanov**

### CODE REVIEW

File: haskell-load.el [hunk 1]
```diff
@@ -88,7 +88,14 @@ Restarts the SESSION's process if that is the case."
    ((haskell-process-consume process "\nBuilding \\(.+?\\)\\.\\.\\.")
     (let ((msg (format "Building: %s" (match-string 1 buffer))))
       (haskell-interactive-mode-echo (haskell-process-session process) msg)
-      (haskell-mode-message-line msg)))))
+      (haskell-mode-message-line msg)))
+   ((string-match "\\(Collecting type info for [[:digit:]]+ module(s) ...\\)"
```
Comment (gracjan) [inline haskell-load.el]: Please remove the `\\(` and `\\)` and use `string-match 0` instead of `string-match 1` later on.

**Comment by gracjan** [comment]
Overall it is a shame that we do not have any automated tests for this kind of stuff. Next time a GHC 8.0.x or 9.x comes out it may change this string and functionality will silently fail.

**Comment by geraldus** [comment]
This is second time we've beaten by changed GHC output.  You're right, we should have tests for all  GHC output parsing code pieces.

**Comment by vshabanov** [comment]
I've made changes you've proposed.

**Pull request merged**